shell/server/proxy.js

```diff
@@ -310,14 +321,27 @@ function openSessionInternal(grainId, user, title, apiToken) {
 
   updateLastActive(grainId, userId);
 
-  var isOwner = grainInfo.owner === userId;
+  cachedSalt = cachedSalt || Random.id(22);
+  var sessionId = generateSessionId(grainId, userId, cachedSalt);
+  var session = Sessions.findOne({_id: sessionId});
```

Comment:
It looks like all anonymous sessions to a single grain are sharing the same entry in the `Sessions` collection. What if there are many tokens, each with a different set of permissions? And happens when a token gets revoked?